libsolidity/codegen/ExpressionCompiler.cpp

```diff
@@ -881,9 +881,14 @@ bool ExpressionCompiler::visit(MemberAccess const& _memberAccess)
 					// We do not visit the expression here on purpose, because in the case of an
 					// internal library function call, this would push the library address forcing
 					// us to link against it although we actually do not need it.
-					auto const* function = dynamic_cast<FunctionDefinition const*>(_memberAccess.annotation().referencedDeclaration);
-					solAssert(!!function, "Function not found in member access");
-					m_context << m_context.functionEntryLabel(*function).pushTag();
+					if (auto const* function = dynamic_cast<FunctionDefinition const*>(_memberAccess.annotation().referencedDeclaration))
+						m_context << m_context.functionEntryLabel(*function).pushTag();
+					else if (dynamic_cast<EventDefinition const*>(_memberAccess.annotation().referencedDeclaration))
```

Comment:
Yes, location turned into the wrong name in the meantime. It should rather be called "function type" or "how to invoke it".